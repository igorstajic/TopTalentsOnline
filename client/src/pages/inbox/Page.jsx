//@flow
import React, { useState, useEffect } from 'react';
import { withSnackbar } from 'notistack';

import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Divider from '@material-ui/core/Divider';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import DeleteIcon from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';

import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import Link from '@material-ui/core/Link';
import moment from 'moment';

import axios from '../../configs/axios';
import LoadingIndicator from '../../layout/LoadingIndicator';

const useStyles = makeStyles(theme => ({
  list: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
    marginTop: theme.spacing(4),
  },
  maxWidth: 360,
  paper: {
    padding: theme.spacing(8),
    marginTop: theme.spacing(5),
    marginBottom: theme.spacing(5),
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  inline: {
    display: 'inline',
  },
  noMessages: {
    marginTop: theme.spacing(5),
    textAlign: 'center',
  },
}));

function Inbox({ enqueueSnackbar, uid }: { uid: string, enqueueSnackbar: Function }) {
  const classes = useStyles();
  const [messages, setMessages] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    async function getMessages() {
      try {
        const getMessagesResponse = await axios.get(`/messages/for-user/${uid}`);
        setMessages(getMessagesResponse.data.messages);
        setIsLoading(false);
      } catch (error) {
        enqueueSnackbar(String(error.response.data.details || 'Something went wrong!'), { variant: 'error' });
      }
    }
    getMessages();
  }, [enqueueSnackbar, uid]);

  const handleDelete = async messageID => {
    try {
      await axios.delete(`/messages/${messageID}`);
      setMessages(messages.filter(m => m.id !== messageID));
    } catch (error) {
      enqueueSnackbar(String(error.response.data.details || 'Something went wrong!'), { variant: 'error' });
    }
  };
  return (
    <Container maxWidth="md" component="section">
      {isLoading ? (
        <LoadingIndicator />
      ) : (
        <Paper className={classes.paper}>
          <Typography component="h1" variant="h4">
            {`My Messages`}
          </Typography>
          {messages.length > 0 ? (
            <List className={classes.list}>
              {messages.map((messageItem, idx) => (
                <>
                  <ListItem alignItems="flex-start">
                    <ListItemAvatar>
                      <Avatar>{`${messageItem.contactName.split(' ').reduce((initials, word) => `${initials}${word[0]}`, '')}`}</Avatar>
                    </ListItemAvatar>
                    <ListItemText
                      primary={contactDetails({ name: messageItem.contactName, email: messageItem.from })}
                      secondary={
                        <>
                          <Typography component="span" variant="body2" className={classes.inline} color="textPrimary">
                            {moment(messageItem.timestamp).format('DD MMM, YYYY HH:mm')}
                          </Typography>
                          {` â€” ${messageItem.message}`}
                        </>
                      }
                    />
                    <ListItemSecondaryAction>
                      <IconButton edge="end" aria-label="Delete" onClick={() => handleDelete(messageItem.id)}>
                        <DeleteIcon />
                      </IconButton>
                    </ListItemSecondaryAction>
                  </ListItem>
                  {idx < messages.length - 1 && <Divider variant="inset" component="li" />}
                </>
              ))}
            </List>
          ) : (
            <Typography component="span" variant="body1" className={classes.noMessages} color="textSecondary">
              No messages yet.
            </Typography>
          )}
        </Paper>
      )}
    </Container>
  );
}

const contactDetails = ({ name, email }) => (
  <span>
    <Typography>
      {`${name} `}
      <Link href={`mailto:${email}`} rel="noopener">
        {`<${email}>`}
      </Link>
    </Typography>
  </span>
);

export default withSnackbar(Inbox);
