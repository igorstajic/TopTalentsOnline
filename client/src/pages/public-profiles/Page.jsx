import React, { useState, useEffect } from 'react';

import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';

import { withSnackbar } from 'notistack';

import axios from '../../configs/axios';
import LoadingIndicator from '../../layout/LoadingIndicator';

import Filters from './Filters';
import List from './CardList';

function PublicProfiles({ enqueueSnackbar }) {
  const [isLoading, setIsLoading] = useState(true);
  const [allProfiles, setAllProfiles] = useState([]);
  const [filteredProfiles, setFilteredProfiles] = useState([]);

  useEffect(() => {
    async function getProfileData() {
      try {
        const getProfilesResponse = await axios.get(`/users/`);
        setAllProfiles(getProfilesResponse.data.users);
        setIsLoading(false);
      } catch (error) {
        enqueueSnackbar(String(error.response.data.details || 'Something went wrong!'), { variant: 'error' });
      }
    }
    getProfileData();
  }, [enqueueSnackbar]);

  return (
    <Container maxWidth="lg" component="section">
      {isLoading ? (
        <LoadingIndicator />
      ) : (
        <Grid container spacing={2}>
          <Grid item xs={12}>
            <Filters allProfiles={allProfiles} setFilteredProfiles={setFilteredProfiles} />
          </Grid>
          <Container maxWidth="md" component="section">
            <List profiles={filteredProfiles} />
          </Container>
        </Grid>
      )}
    </Container>
  );
}

export default withSnackbar(PublicProfiles);
