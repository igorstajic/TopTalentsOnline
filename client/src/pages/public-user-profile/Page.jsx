import React, { useState, useEffect } from 'react';

import Container from '@material-ui/core/Container';
import Chip from '@material-ui/core/Chip';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';

import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';

import LocationIcon from '@material-ui/icons/LocationOn';
import MessageIcon from '@material-ui/icons/Message';

import { withSnackbar } from 'notistack';

import axios from '../../configs/axios';
import LoadingIndicator from '../../layout/LoadingIndicator';
import ContactForm from '../../components/ContactForm';

import { makeLocationString } from '../../helpers/formatters';
const useStyles = makeStyles(theme => ({
  paper: {
    padding: theme.spacing(8),
    marginTop: theme.spacing(5),
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  chip: {
    margin: theme.spacing(0.25),
  },
  category: {
    textTransform: 'capitalize',
  },
  label: {
    display: 'flex',
    alignItems: 'center',
  },
}));

function PublicUserProfile({ match, enqueueSnackbar }) {
  const [isLoading, setIsLoading] = useState(true);
  const [profile, setProfile] = useState(null);
  const [isShowingContactForm, showContactForm] = useState(false);

  const classes = useStyles();
  useEffect(() => {
    async function getProfileData() {
      try {
        const getProfileResponse = await axios.get(`/users/${match.params.uid}`);
        setProfile(getProfileResponse.data.user);
        setIsLoading(false);
      } catch (error) {
        enqueueSnackbar(String(error.response.data.details || 'Something went wrong!'), { variant: 'error' });
      }
    }
    getProfileData();
  }, [enqueueSnackbar, match.params.uid]);

  return (
    <Container maxWidth="md" component="section" data-testid="page_container__publicProfile">
      {isLoading ? (
        <LoadingIndicator />
      ) : (
        <Paper className={classes.paper}>
          <Typography component="h1" variant="h3">
            {`${profile.firstName} ${profile.lastName}`}
          </Typography>
          <Typography className={classes.category} component="h2" variant="subtitle1">
            {profile.category}
          </Typography>

          <Box mt={4} mb={4}>
            <Typography className={classes.label} component="p" variant="body1">
              {(profile.city || profile.country) && <LocationIcon color="action" />} {makeLocationString(profile.city, profile.country)}
            </Typography>
          </Box>

          <Box mt={2}>
            {profile.subCategories.length > 0 && <Typography component="h4">Skills:</Typography>}
            {profile.subCategories.map((skill, idx) => (
              <Chip className={classes.chip} color="secondary" key={`${skill}_${idx}`} label={skill} />
            ))}
          </Box>

          <Box mt={8}>
            <Button variant="contained" color="primary" size="small" onClick={() => showContactForm(true)}>
              <Box px={2} py={1} display="flex">
                <MessageIcon color="inherit" />
                <Box ml={1}>contact</Box>
              </Box>
            </Button>
          </Box>
        </Paper>
      )}
      <ContactForm uid={match.params.uid} isOpen={isShowingContactForm} handleClose={() => showContactForm(false)} />
    </Container>
  );
}

export default withSnackbar(PublicUserProfile);
