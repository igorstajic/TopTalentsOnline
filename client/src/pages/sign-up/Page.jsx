//@flow
import React, { useState, useEffect, useCallback, useContext } from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Fade from '@material-ui/core/Fade';
import { withSnackbar } from 'notistack';

import LockOutlinedIcon from '@material-ui/icons/LockOutlined';

import RouterLink from '../../components/RouterLink';
import axios from '../../configs/axios';

import { emailValidatior } from '../../helpers/validators';
import { SessionContext, authenticateSession } from '../../services/session';

const useStyles = makeStyles(theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

function SignUp({ history, enqueueSnackbar }: { history: Object, enqueueSnackbar: Function }) {
  const classes = useStyles();
  const session = useContext(SessionContext);

  const [firstName, setFirstName] = useState('');
  const [firstNameErrorMessage, setFirstNameErrorMessage] = useState('');

  const [lastName, setLastName] = useState('');
  const [lastNameErrorMessage, setLastNameErrorMessage] = useState('');

  const [email, setEmail] = useState('');
  const [emailErrorMessage, setEmailErrorMessage] = useState('');

  const [password, setPassword] = useState('');
  const [passwordErrorMessage, setPasswordErrorMessage] = useState('');

  const [passwordConfirmation, setPasswordConfirmation] = useState('');
  const [passwordConfirmationErrorMessage, setPasswordConfirmationErrorMessage] = useState('');
  const [isSubmitted, setIsSubmitted] = useState(false);

  const validateInputs = useCallback(() => {
    let isValid = false;

    if (firstName.trim().length === 0) {
      setFirstNameErrorMessage('First name is required');
    } else {
      setFirstNameErrorMessage('');
      isValid = true;
    }
    if (lastName.trim().length === 0) {
      setLastNameErrorMessage('Last name is required');
    } else {
      setLastNameErrorMessage('');
      isValid = true;
    }
    if (email.trim().length === 0) {
      setEmailErrorMessage('Email is required');
    } else if (!emailValidatior.test(email)) {
      setEmailErrorMessage('Email is invalid');
    } else {
      setEmailErrorMessage('');
      isValid = true;
    }

    if (password.trim().length === 0) {
      setPasswordErrorMessage('Password is required');
      isValid = false;
    } else {
      setPasswordErrorMessage('');
    }

    if (password.trim() !== passwordConfirmation.trim()) {
      setPasswordConfirmationErrorMessage('Passwords must match');
      isValid = false;
    } else {
      setPasswordConfirmationErrorMessage('');
    }

    return isValid;
  }, [email, firstName, lastName, password, passwordConfirmation]);

  useEffect(() => {
    if (isSubmitted) {
      validateInputs();
    }
  }, [isSubmitted, validateInputs]);

  const onSubmit = async event => {
    setIsSubmitted(true);
    event.preventDefault();

    if (validateInputs()) {
      try {
        await axios.post('/users/register', {
          email,
          password,
          firstName,
          lastName,
        });
        const authenticationResponse = await authenticateSession(email, password);
        if (authenticationResponse.user) {
          session.setCurrentUser(authenticationResponse.user);
          history.push('/edit-profile');
        } else {
          enqueueSnackbar(authenticationResponse.error, { variant: 'error' });
        }
        enqueueSnackbar('Welcome to TopTalents Online', { variant: 'success' });
      } catch (error) {
        enqueueSnackbar(error.response.data.details || 'Something went wrong!', { variant: 'error' });
      }
    }
  };

  return (
    <Fade in={true}>
      <Container component="section" maxWidth="xs">
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign up
          </Typography>
          <form className={classes.form} noValidate onSubmit={onSubmit}>
            <Grid container spacing={2}>
              <Grid item xs={12} sm={6}>
                <TextField
                  name="firstName"
                  variant="outlined"
                  required
                  fullWidth
                  id="firstName"
                  label="First Name"
                  autoFocus
                  error={firstNameErrorMessage.length > 0}
                  helperText={firstNameErrorMessage}
                  value={firstName}
                  onChange={ev => setFirstName(ev.target.value)}
                />
              </Grid>
              <Grid item xs={12} sm={6}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="lastName"
                  label="Last Name"
                  name="lastName"
                  error={lastNameErrorMessage.length > 0}
                  helperText={lastNameErrorMessage}
                  value={lastName}
                  onChange={ev => setLastName(ev.target.value)}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  error={emailErrorMessage.length > 0}
                  helperText={emailErrorMessage}
                  value={email}
                  onChange={ev => setEmail(ev.target.value)}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  error={passwordErrorMessage.length > 0}
                  helperText={passwordErrorMessage}
                  value={password}
                  onChange={ev => setPassword(ev.target.value)}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  name="passwordConfirmation"
                  label="Confirm Password"
                  type="password"
                  id="password_confirmation"
                  error={passwordConfirmationErrorMessage.length > 0}
                  helperText={passwordConfirmationErrorMessage}
                  value={passwordConfirmation}
                  onChange={ev => setPasswordConfirmation(ev.target.value)}
                />
              </Grid>
            </Grid>
            <Button type="submit" fullWidth variant="contained" color="primary" className={classes.submit}>
              Sign Up
            </Button>
            <Grid container justify="flex-end">
              <Grid item>
                <Link component={RouterLink} to="/login" variant="body2">
                  Already have an account? Sign in
                </Link>
              </Grid>
            </Grid>
          </form>
        </div>
      </Container>
    </Fade>
  );
}

export default withSnackbar(SignUp);
