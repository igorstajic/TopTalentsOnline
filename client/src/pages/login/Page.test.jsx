// Link.react.test.js
import '@babel/polyfill';

import React from 'react';
import { render } from '@testing-library/react';

import { SnackbarProvider } from 'notistack';

import { BrowserRouter as Router } from 'react-router-dom';

import Page from './Page';

function renderWithRouter(ui) {
  return {
    ...render(
      <SnackbarProvider>
        <Router>{ui}</Router>
      </SnackbarProvider>
    ),
  };
}

describe('Render Login Page', () => {
  it('Renders the component', async () => {
    const { container } = renderWithRouter(<Page />);

    expect(container).toMatchSnapshot();
  });
});
