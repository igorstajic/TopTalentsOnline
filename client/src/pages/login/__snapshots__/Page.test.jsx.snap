// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Page Renders the component 1`] = `
<div>
  <section
    class="MuiContainer-root MuiContainer-maxWidthXs"
    data-testid="page_container__login"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="makeStyles-paper-1"
    >
      <div
        class="MuiAvatar-root makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <g
            fill="none"
          >
            <path
              d="M0 0h24v24H0V0z"
            />
            <path
              d="M0 0h24v24H0V0z"
              opacity=".87"
            />
          </g>
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <form
        class="makeStyles-form-3"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
            data-shrink="true"
            for="email"
          >
            Email Address
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-108 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-109"
                style="width: 0px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
            <input
              aria-invalid="false"
              autocomplete="email"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="input__email"
              id="email"
              name="email"
              required=""
              type="email"
              value=""
            />
          </div>
          
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="false"
            for="password"
          >
            Password
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-108 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-109"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
            <input
              aria-invalid="false"
              autocomplete="current-password"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="input__password"
              id="password"
              name="password"
              required=""
              type="password"
              value=""
            />
          </div>
          
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-submit-4 MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          data-testid="action__submit"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Sign In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              data-testid="link__signup"
              href="/sign-up"
            >
              Don't have an account? Sign Up
            </a>
          </div>
        </div>
      </form>
    </div>
  </section>
</div>
`;
